app/src/main/java/com/github/se/eventradar/ui/component/ReusableComponents.kt

```diff
@@ -144,162 +146,170 @@ fun EventCard(event: Event, onCardClick: (String) -> Unit) {
+                  modifier = Modifier.testTag("freeSwitch"))
             }
 
-        Spacer(modifier = Modifier.height(4.dp))
-
         // Buttons for category selection
         Text(
-            text = "Category: ",
-            modifier = Modifier.weight(1f),
+            text = stringResource(id = R.string.category_label),
+            modifier = Modifier.weight(1f).testTag("categoryLabel"),
             style =
                 TextStyle(
                     fontSize = 16.sp,
                 ))
 
         Row(
             verticalAlignment = Alignment.CenterVertically,
-            horizontalArrangement = Arrangement.Start) {
-              CategorySelection()
-            }
+            horizontalArrangement = Arrangement.Start,
+            modifier = Modifier.testTag("categoryRow"),
+        ) {
+          CategorySelection(
+              uiState = uiState,
+              modifier = Modifier.testTag("categoryOptionsColumn"),
+          )
+        }
 
         Spacer(modifier = Modifier.height(4.dp))
 
         // Button to apply filter
-        Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.Center) {
-          Button(
-              onClick = {
-                // Apply filter
-                onRadiusChange(uiState.radiusInputFilter)
-                onFreeSelectionChange(uiState.freeEventsFilter)
-                onCategorySelectionChange(uiState.categorySelectionFilter)
-                onCategorySelectionChange(uiState.categorySelectionFilter)
-              }) {
-                Text("Apply")
-              }
-        }
+        Row(
+            modifier = Modifier.fillMaxWidth().testTag("filterApplyRow"),
+            horizontalArrangement = Arrangement.Center) {
+              Button(
+                  onClick = { onFilterApply() }, modifier = Modifier.testTag("filterApplyButton")) {
+                    Text(stringResource(id = R.string.filter_apply))
+                  }
+            }
       }
     }
   }
 }
 
+// TO DO: Need to convert category selection to MVVM and test
 @Composable
-fun CategorySelection() {
-  LazyColumn {
+fun CategorySelection(uiState: EventsOverviewUiState, modifier: Modifier) {
+  LazyColumn(modifier = modifier) {
     items(EventCategory.entries) { category ->
-      var isChecked by remember { mutableStateOf(false) }
+      var isChecked by remember { mutableStateOf(uiState.categoriesCheckedList.contains(category)) }
       Row(
           verticalAlignment = Alignment.CenterVertically,
           horizontalArrangement = Arrangement.Start,
-          modifier = Modifier.padding(vertical = 0.dp)) {
+          modifier = Modifier.padding(vertical = 0.dp).testTag("categoryRow")) {
             Checkbox(
                 checked = isChecked,
-                onCheckedChange = { isChecked = it },
-                modifier = Modifier.scale(0.6f).size(10.dp).padding(start = 10.dp))
+                onCheckedChange = {
```

Comment:
Would be nice to add a test covering this, but other than that, LGTM!

Reply:
I have already spent a few hours trying to test this part but with no success. Since test coverage is above 80% and testing this feature on the emulator seems to work already, I think it's more important to get this PR merged so I can start spending my time on other features (delivering value) even if this PR is not perfect, rather than focusing my time on testing every single part of the code.